addons/hr_timesheet/report/report_timesheet_templates.xml

```diff
@@ -1,4 +1,64 @@
 <odoo>
+    <template id="hr_timesheet.timesheet_table">
+        <div class="row" style="margin-top:10px;">
+            <div class="col-lg-12">
+                <table class="table table-sm">
+                    <thead>
+                        <tr>
+                            <th class="align-middle"><span>Date</span></th>
+                            <th class="align-middle"><span>Responsible</span></th>
+                            <th class="align-middle"><span>Description</span></th>
+                            <th class="align-middle" t-if="show_project"><span>Project</span></th>
+                            <th class="align-middle" t-if="show_task"><span>Task</span></th>
+                            <th class="text-right">
+                                <span t-if="is_uom_day">Time Spent (Days)</span>
+                                <span t-else="">Time Spent (Hours)</span>
+                            </th>
+                        </tr>
+                   </thead>
+                   <tbody>
+                        <tr t-foreach="docs" t-as="line">
```

Comment:
You may use `lines` rather than `docs` in order to avoid ambiguity between variables.